pulsar-io/kafka-connect-adaptor/src/main/java/org/apache/pulsar/io/kafka/connect/KafkaConnectSource.java

```diff
@@ -95,8 +134,17 @@ private void initTransforms(Map<String, Object> config) {
                             java.util.Map.Entry::getValue
                         ));
                     log.info("transform config: {}", transformConfig);
+                    String predicateName = (String) transformConfig.get("predicate");
+                    Predicate<SourceRecord> predicate = null;
+                    if (predicateName != null) {
+                        predicate = predicates.get(predicateName);
+                        if (predicate == null) {
+                            log.warn("Transform {} references non-existent predicate: {}",
+                                    transformName, predicateName);
+                        }
+                    }
                     transform.configure(transformConfig);
-                    transformations.add(transform);
+                    transformations.add(Pair.of(predicate, transform));
```

Comment:
In addition, there would be a need to support `negate` as documented [in the SMT predicate docs](https://kafka.apache.org/documentation/#connect_predicates): "The transformation's implicit `negate` config properties allows us to invert the set of records which a predicate matches.".

Instead of a `Pair`, it's better to use a Java record to hold information.

Reply:
Thanks for holding hands with me. As an Elixir developer who hasn't touched Java in years, I genuinely appreciate the time you are investing on this :pray: I should have something ready soon, and I will also add a couple of unit tests to cover predicates and negated predicates.

Reply:
Good work so far!